source/OptimaJet.DWKit.StarterApplication/Repositories/ProjectRepository.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using JsonApiDotNetCore.Data;
+using JsonApiDotNetCore.Services;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Optimajet.DWKit.StarterApplication.Models;
+using OptimaJet.DWKit.StarterApplication.Services;
+using Serilog;
+
+namespace OptimaJet.DWKit.StarterApplication.Repositories
+{
+    public class ProjectRepository : DefaultEntityRepository<Project>
+    {
+        public IOrganizationContext OrganizationContext { get; }
+        public ICurrentUserContext CurrentUserContext { get; }
+        public UserRepository UserRepository { get; set; }
+
+        public ProjectRepository(
+            ILoggerFactory loggerFactory,
+            IJsonApiContext jsonApiContext,
+            IOrganizationContext organizationContext,
+            ICurrentUserContext currentUserContext,
+            UserRepository userRepository,
+            IDbContextResolver contextResolver
+            ) : base(loggerFactory, jsonApiContext, contextResolver)
+        {
+            this.OrganizationContext = organizationContext;
+            this.CurrentUserContext = currentUserContext;
+            this.UserRepository = userRepository;
+        }
+
+        public override IQueryable<Project> Get()
+        {
+            if (OrganizationContext.SpecifiedOrganizationDoesNotExist) return Enumerable.Empty<Project>().AsQueryable();
```

Comment:
since a lot of this behavior may be duplicated between the controllers, I wonder if it'd be worth it to pull out into a BaseRepository or some other shared context that could evaluate most of this user / org logic. 

Maybe the api could be:

```cs
// always returns an array
var orgIds = GetCurrentOrganizationIds();

return base.Get()
           .Where(p => orgIds.Contains(p.OrganizationId)
```

off hand, I don't know if this would have any performance impacts, if any, but I think it could reduce the amount of code we have to maintain. :)

Reply:
Added `User.OrganizationIds`